adapters/handlers/rest/configure_api.go

```diff
@@ -201,13 +204,36 @@ func MakeAppState(ctx context.Context, options *swag.CommandLineOptionsGroup) *s
 	appState := startupRoutine(ctx, options)
 
 	if appState.ServerConfig.Config.Monitoring.Enabled {
-		appState.ServerMetrics = monitoring.NewServerMetrics(appState.ServerConfig.Config.Monitoring, prometheus.DefaultRegisterer)
+		appState.ServerMetrics = monitoring.NewServerMetrics(appState.ServerConfig.Config.Monitoring.MetricsNamespace, prometheus.DefaultRegisterer)
 		appState.TenantActivity = tenantactivity.NewHandler()
 
 		// export build tags to prometheus metric
 		build.SetPrometheusBuildInfo()
 		prometheus.MustRegister(version.NewCollector(build.AppName))
 
+		opts := armonprometheus.PrometheusOpts{
+			Expiration: 0, // never expire any metrics,
```

Comment:
Could I get some help understanding this? I read the doc for `Expiration` but don't totally get it: https://pkg.go.dev/github.com/armon/go-metrics/prometheus#PrometheusOpts . If `Expiration=0`, then does that mean Weaviate will keep all metrics until they've been scraped by Prometheus. If `Expiration=1min`, does that mean Weaviate will drop any metrics which aren't queried within one minutes of being set? Or is this related to retention on the Prometheus side? https://prometheus.io/docs/prometheus/latest/storage/

Reply:
> If Expiration=0, then does that mean Weaviate will keep all metrics until they've been scraped by Prometheus

I don't think this expiration have anything to do with prometheus scraping (we use pull based)

> does that mean Weaviate will drop any metrics which aren't queried within one minutes of being set?

Yes it's all on application side, we may drop this metrics before ending up in prometheus server.

> Or is this related to retention on the Prometheus side? https://prometheus.io/docs/prometheus/latest/storage/

Prometheus retention is something different this expiration value. Unlike databases like Weaviate, Time series database (like Prometheus) can delete data that is say older than 30 days. That 30 days is retention value. And querying will touch only latest 30 days data.